api/RestfulService.php

```diff
@@ -178,18 +184,11 @@ public function request($subURL = '', $method = "GET", $data = null, $headers =
 			$curlOptions + (array)$this->config()->default_curl_options,
 			$this->getBasicAuthString()
 		));
-
-		// Check for unexpired cached feed (unless flush is set)
-		//assume any cache_expire that is 0 or less means that we dont want to
-		// cache
-		if($this->cache_expire > 0 && self::$flush
-				&& @file_exists($cache_path)
-				&& @filemtime($cache_path) + $this->cache_expire > time()) {
-
-			$store = file_get_contents($cache_path);
-			$response = unserialize($store);
-
-		} else {
+		$cache = SS_Cache::factory("RestfulService", 'Output', array(
+			'lifetime'=>$this->cache_expire,
+			'automatic_serialization'=> true
+		));
+		if (!$response = $cache->load($cacheKey) || self::$flush){
```

Comment:
This is going to break as the variable assignment here is equivelent to:

```php
$response = ($cache->load($cacheKey) || self::$flush)
```

when you want:

```php
($response = $cache->load($cacheKey)) || self::$flush)
```